internal/datasource/nginx/instances.go

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package nginx
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/nginx/agent/v3/internal/model/instances"
+	"github.com/nginx/agent/v3/internal/model/os"
+	"github.com/nginx/agent/v3/internal/util"
+)
+
+var (
+	re     = regexp.MustCompile(`(?P<name>\S+)/(?P<version>\S+)`)
+	plusre = regexp.MustCompile(`(?P<name>\S+)/(?P<version>\S+).\((?P<plus>\S+plus\S+)\)`)
+)
+
+type GetInfo func(pid int32, exe string) (*NginxInfo, error)
+
+type Nginx struct {
+	parameters NginxParameters
+}
+
+type NginxParameters struct {
+	GetInfo GetInfo
+}
+
+func NewNginx(parameters NginxParameters) *Nginx {
```

Comment:
nit: https://google.github.io/styleguide/go/decisions.html#package-vs-exported-symbol-name
```suggestion
func New(parameters NginxParameters) *Nginx {
```
So when the caller inits this data source, it'll look like this
```
nginx := nginx.New(params)
```
instead of
```
nginx := nginx.NewNginx(params)
```

Reply:
After looking at `process.go`, it looks like it should be something like `NewInstance` or something like that.

Reply:
Thanks for the feedback. I'll take a look at refactoring this.